plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/EditorProductPicker.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useBlockProps } from '@wordpress/block-editor';
+import { Icon, info } from '@wordpress/icons';
+import ProductControl from '@woocommerce/editor-components/product-control';
+import {
+	Placeholder,
+	// @ts-expect-error Using experimental features
+	__experimentalHStack as HStack,
+	// @ts-expect-error Using experimental features
+	__experimentalText as Text,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import type { ProductCollectionEditComponentProps } from '../types';
+import { getCollectionByName } from '../collections';
+
+const EditorProductPicker = ( props: ProductCollectionEditComponentProps ) => {
+	const blockProps = useBlockProps();
+	const attributes = props.attributes;
+
+	const collection = getCollectionByName( props.attributes.collection );
+	if ( ! collection ) {
+		return;
+	}
+
+	return (
+		<div { ...blockProps }>
+			<Placeholder className="wc-blocks-product-collection__editor-product-picker">
+				<HStack alignment="center">
+					<Icon
+						icon={ info }
+						className="wc-blocks-product-collection__info-icon"
+					/>
+					<Text>
+						<strong>{ collection.title }</strong> requires a product
+						to be selected in order to display associated items.
+					</Text>
+				</HStack>
+				<ProductControl
+					selected={ attributes.selectedReference?.id || 0 }
+					showVariations
+					onChange={ ( value = [] ) => {
+						const id = value[ 0 ] ? value[ 0 ].id : 0;
+						props.setAttributes( {
+							selectedReference: {
+								type: 'product',
+								id,
+							},
+						} );
```

Comment:
I know it's very unlikely but if we cover a case `value` may not be available and we fallback to some value, I don't think we should be setting a `selectedReference.type` to `product`. What I mean is I'd expect a logic like:

```suggestion
					onChange={ ( value = [] ) => {
						const id = value[ 0 ] ? value[ 0 ].id : null;
						
						if (id) {
						    props.setAttributes( {
							    selectedReference: {
								    type: 'product',
								    id,
							    },
						    } );
						}
```

I can't think of a specific use case in which it could hurt us but I guess with collection allowing multiple contexts, overriding `selectedReference` when for some reason we don't get product ID may be erroneous. But I may be overcomplicating it as well 😅

Reply:
Good point! I have [made the changes](https://github.com/woocommerce/woocommerce/blob/d06d591cac386535b2d0f55eb72920ed3c14177a/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/EditorProductPicker.tsx#L50-L60). Does that look good to you? 🙂